src/store/actions.js

```diff
@@ -406,7 +406,18 @@ export default {
 			return envelopes
 		})
 	},
-	syncEnvelopes({ commit, getters, dispatch }, { mailboxId, query, init = false }) {
+	syncEnvelopesByRole({ commit, getters, dispatch }, { role, accountId, query, init = false, criteria = 'all' }) {
+		const mailboxes = getters.getMailboxes(accountId)
```

Comment:
```suggestion
		getters.getMailboxes(accountId)
		    .filter(mb => mb.specialRole === role)
		    .map(mb => dispatch(...))
```

for short and concise code :wink: